php/class-post-type.php

```diff
@@ -1331,4 +1409,134 @@ public function handle_snapshot_fork() {
+				$input .= checked( $value['selected_uuid'], $conflicted_data['uuid'], false ) . '>';
+				echo $input; // WPCS: xss ok.
+				echo '<code> ' . esc_html( $conflicted_data['uuid'] ) . ' </code></summary>';
+
+				$preview = $this->get_printable_setting_value( $conflicted_data['value'] );
+
+				echo '<div class="snapshot-conflict-setting-data">';
+				echo $preview; // WPCS: xss ok.
+				echo '</div>';
+				echo '</details></input></li>';
+			}
+			echo '</ul>';
+			echo '</div>';
+		} // End if().
+	}
+
+	/**
+	 * Change conflicted value in post_content, if they were changed in the meta box.
+	 *
+	 * @param String $content Post content to filter.
+	 * @return String $content Post content, possibly filtered.
+	 */
+	public function filter_selected_conflict_setting( $content ) {
+		global $post;
+		$key_for_selected_resolved_setting = static::SLUG . '_resolve_conflict_uuid';
+		$post_type_object = get_post_type_object( static::SLUG );
+
+		$should_filter_for_resolve_conflict = (
+			isset( $post->post_status )
+			&&
+			( 'publish' !== $post->post_status )
+			&&
+			current_user_can( $post_type_object->cap->edit_post, $post->ID )
+			&&
+			( static::SLUG === $post->post_type )
+			&&
+			! empty( $_REQUEST[ $key_for_selected_resolved_setting ] )
+			&&
+			is_array( $_REQUEST[ $key_for_selected_resolved_setting ] )
+			&&
+			isset( $_REQUEST[ static::SLUG ] )
+			&&
+			wp_verify_nonce( $_REQUEST[ static::SLUG ], static::SLUG . '_settings' )
+			&&
+			! ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE )
+		);
+
+		if ( ! $should_filter_for_resolve_conflict ) {
+			return $content;
+		}
+		$setting_ids_to_unset = wp_unslash( $_REQUEST[ $key_for_selected_resolved_setting ] );
+		$snapshot_data = json_decode( wp_unslash( $content ), true );
+		foreach ( $setting_ids_to_unset as $i => $data ) {
+			$data = json_decode( $data, true );
+			if ( isset( $data['setting_id'], $data['uuid'], $snapshot_data[ $data['setting_id'] ], $snapshot_data[ $data['setting_id'] ]['merge_conflict'] ) ) {
+				$setting_key = $data['setting_id'];
+				if ( isset( $snapshot_data[ $setting_key ]['selected_uuid'] ) && $snapshot_data[ $setting_key ]['selected_uuid'] === $data['uuid'] ) {
+					// No change.
+					continue;
+				}
+				$snapshot_data[ $setting_key ]['selected_uuid'] = $data['uuid'];
+				foreach ( $snapshot_data[ $setting_key ]['merge_conflict'] as $conflict_data ) {
+					if ( isset( $conflict_data['uuid'], $conflict_data['value'] ) && $conflict_data['uuid'] === $data['uuid'] ) {
+						$snapshot_data[ $setting_key ]['value'] = $conflict_data['value'];
+						break;
+					}
+				}
+			}
+		}
+		$content = Customize_Snapshot_Manager::encode_json( $snapshot_data );
+		return $content;
+	}
+
+	/**
+	 * Get snapshot merged uuid from post_content
+	 *
+	 * @param array $post_content snapshot data.
+	 * @return array merged snapshot.
+	 */
+	public function get_snapshot_merged_uuid( $post_content ) {
```

Comment:
Maybe this should be `get_merged_changesets()`. Let's also make it `private` as I don't see a need for it to be exposed.

Reply:
Yep, will update in #93.

Reply:
```php
	/**
	 * Get snapshot merged uuid from post_content
	 *
	 * @internal
	 *
	 * @param array $post_content snapshot data.
	 * @return array merged snapshot.
	 */
	public function _get_merged_changesets( $post_content ) {
```
@westonruter How about this? 
If we make this private we need to mock that method in unit-test which is not best practice. We can cover this method under render function's unit-test.